Feature/UA-401 Indentation of measurements in a DL

kotodharma opened the pull request:
Indentation of measurements in a DL

jonpage commented on app/controllers/data_lists_controller.rb:
```diff
@@ -268,10 +272,26 @@ def remove_measurement
     def data_list_params
       params.require(:data_list)
-          .permit(:name, :list, :startyear, :created_by, :updated_by, :owned_by, :measurements, :measurement_id)
+          .permit(:name, :list, :startyear, :created_by, :updated_by, :owned_by, :measurements, :measurement_id, :indded)
```
What is `indded`? Can we use more descriptive names? It will make our codebase more maintainable. The same comment applies to `curind`, `newind`. I can live with the abbreviated model name, since we use that pattern liberally throughout the code base, but otherwise we should limit abbreviation. Alternatives in this case could be `current_indent`, `new_indent`. I can't figure out what the long form of `indded` would be.

jonpage commented on app/helpers/data_lists_helper.rb:
```diff
@@ -67,16 +67,11 @@ def dl_linked_version(description)
-end
 
+  def make_indentation(n)
+    ## HTML string below needs to keep internal quotes as single, because it gets embedded in JSON
```
It would be a bit more idiomatic for us to return something like `{"intent": 1}` as our JSON, then use that number to create the HTML in the CoffeeScript file.

jonpage replied:
We could still use this helper in the view, but returning raw HTML in as a JSON value is usually frowned upon.

jonpage replied:
We can leave this as is. We have probably been through enough iterations on this piece of code.

kotodharma commented:
OK, I think I've addressed the concerns. Note that I avoided naming some things just "indent", which might have been concise and appropriate, just because there would have ended up a variety of different things (variable, POST parameter, model attribute, JSON key, etc) with this name, and I thought that might be confusing to someone (like me :=) reading the code.
